scripts/add_ts_data_bidmatrixinfo.py

```diff
@@ -0,0 +1,154 @@
+
+from typing import Union
+
+from cognite.client.data_classes import TimeSeries, TimeSeriesList
+from cognite.client.exceptions import CogniteNotFoundError
+from cognite.client import CogniteClient
+from cognite.powerops import PowerOpsClient
+
+from cognite.powerops.client._generated.v1.data_classes import BidMatrixInformationWrite
+import random
+import arrow
+
+
+def create_time_series_objects():
+    ts_list = [
+        TimeSeries(external_id="emilie_ts_onhour", name="emilie_ts_onhour", metadata={"data_type": "on_hour"}),
+        TimeSeries(external_id="emilie_ts_empty", name="emilie_ts_empty", metadata={"data_type": "none"}),
+        TimeSeries(external_id="emilie_ts_offhour", name="emilie_ts_offhour", metadata={"data_type": "off_hour"}),
+    ]
+    return ts_list
+
+
+# Function to get existing time series from CDF
+def get_existing_time_series(
+    client: CogniteClient, ts_list: list[TimeSeries]
+) -> tuple[list[TimeSeries], list[TimeSeries]]:
+    want_external_ids = {ts.external_id: ts for ts in ts_list}
+
+    existing_ts_from_cdf = client.time_series.retrieve_multiple(
+        external_ids=list(want_external_ids.keys()), ignore_unknown_ids=True
+    )
+    existing_ts_ext_ids = [ts.external_id for ts in existing_ts_from_cdf]
+
+    existing_ts = [ts for ts in ts_list if ts.external_id in existing_ts_ext_ids]
+    missing_ts = [ts for ts in ts_list if ts.external_id not in existing_ts_ext_ids]
+
+    return existing_ts, missing_ts
+
+
+# Function to create or update time series in CDF
+def create_update_time_series(client: CogniteClient, to_be_updated: list[TimeSeries], missing_ts: list[TimeSeries]):
+    created = []
+    if missing_ts:
+        created = client.time_series.create(missing_ts)
+        if isinstance(created, TimeSeries):
+            created = [created]
+
+    updated = []
+    if to_be_updated:
+        updated = client.time_series.update(to_be_updated)
+        if not isinstance(updated, TimeSeriesList):
+            updated = [updated]
+
+    return created + updated
+
+
+# Function to clean data points for a list of time series
+def clean_data_points(client: CogniteClient, ts_list: list[TimeSeries]):
+    ranges = [{"external_id": ts.external_id, "start": 0, "end": "now"} for ts in ts_list]
+    client.time_series.data.delete_ranges(ranges)
+
+
+# Function to insert data points into according to the type of time series (definined in the metadata)
+def insert_data_points(client: CogniteClient, ts_list: list[TimeSeries]):
+    now = arrow.now().floor("hour")
+    start_time = now.shift(days=-365).timestamp() * 1000
+    end_time = now.timestamp() * 1000
+
+    dps_to_insert = []
+    for ts in ts_list:
+        data_type = ts.metadata.get("data_type", "none")
+
+        if data_type == "on_hour":
+            data_points = [
+                {"timestamp": start_time + i * 3600 * 1000, "value": random.random()}
+                for i in range(int((end_time - start_time) / (3600 * 1000)))
+            ]
```

Comment:
**suggestion:** always best to avoid as much "random math" as possible since it becomes very error-prone given minor typos, so always best to keep things in the most "human recognizable" format if possible and since the [SDK accepts datetime objects](https://cognite-sdk-python.readthedocs-hosted.com/en/latest/time_series.html#insert-datapoints-into-multiple-time-series) then we can use that

```suggestion
    now = arrow.now().floor("hour")
    start_time = now.shift(days=-365)
    end_time = now

    dps_to_insert = []
    for ts in ts_list:
        data_type = ts.metadata.get("data_type", "none")

        if data_type == "on_hour":
            data_points = [
                {"timestamp": start_time.shift(hours=i).datetime, "value": random.random()}
                for i in range(int((end_time - start_time).total_seconds() / 3600))
            ]
```